src/notification.c

```diff
@@ -199,17 +199,29 @@ void notification_free(notification * n)
 }
 
 /*
- * Replace all occurrences of "needle" with a quoted "replacement",
- * according to the markup settings.
+ * Replace the two chars where **needle points
+ * with a quoted "replacement", according to the markup settings.
+ *
+ * The needle is a double pointer and gets updated upon return
+ * to point to the first char, which occurs after replacement.
+ *
  */
-char *notification_replace_format(const char *needle, const char *replacement,
-                                  char *haystack, enum markup_mode markup_mode) {
-        char* tmp;
+char *notification_replace_single_field(char *haystack, char **needle,
+                const char *replacement, enum markup_mode markup_mode) {
+
```

Comment:
Since this is assuming that `**needle` is the beginning of a format flag, I think an assertion is in order to catch any bugs.

Reply:
This is fixed now the line below.

Reply:
Also I added another assertion, that `needle` points into `haystack`.